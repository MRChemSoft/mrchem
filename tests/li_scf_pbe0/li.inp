# vim:syntax=sh:

world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Molecule {
multiplicity = 2
$coords
Li   0.0     0.0      0.0
$end
}

WaveFunction {
  method = PBE0                   # Wave function method (HF or DFT)
  restricted = false
}

SCF {
  kain = 3                        # Length of KAIN iterative history
  max_iter = 5
  orbital_thrs = 2.0e-2
  guess_type = GTO             # Type of initial guess (none, gto, mw)
}
