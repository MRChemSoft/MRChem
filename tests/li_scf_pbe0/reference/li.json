{
  "input": {
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 2
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "li.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "pbe0"
              }
            ],
            "spin": true
          }
        }
      },
      "initial_guess": {
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (PBE0)",
        "prec": 0.001,
        "restricted": false,
        "type": "gto",
        "zeta": 0
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 3,
        "localize": false,
        "max_iter": 5,
        "method": "DFT (PBE0)",
        "orbital_thrs": 0.02,
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        0.0
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 7.281160245956108e-14,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.0
          ],
          "vector_el": [
            0.0,
            0.0,
            0.0
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "Li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 2,
      "orbital_energies": {
        "energy": [
          -2.0565097280178066,
          -0.14104742407372634,
          -2.04544109734709
        ],
        "occupation": [
          1.0,
          1.0,
          1.0
        ],
        "spin": [
          "a",
          "a",
          "b"
        ],
        "sum_occupied": -4.242998249438623
      },
      "scf_energy": {
        "E_ee": 4.048829903319412,
        "E_eext": 0.0,
        "E_el": -7.4663514175705625,
        "E_en": -17.08949877826536,
        "E_kin": 7.381066252257522,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.4663514175705625,
        "E_x": -0.44311604114410696,
        "E_xc": -1.3636327537380306
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 1,
      "routine": "mrchem.x",
      "total_cores": 1,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 4.061917141108662,
        "E_eext": 0.0,
        "E_el": -7.414113655367033,
        "E_en": -17.03503042962955,
        "E_kin": 7.370996808590605,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.414113655367033,
        "E_x": -0.4451123948138887,
        "E_xc": -1.36688478062286
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 4.0354652526630606,
              "E_eext": 0.0,
              "E_el": -7.465540758368596,
              "E_en": -17.012395743212963,
              "E_kin": 7.312117113164178,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": 0.0,
              "E_tot": -7.465540758368596,
              "E_x": -0.4417877046055466,
              "E_xc": -1.3589396763773245
            },
            "energy_total": -7.465540758368596,
            "energy_update": 0.05142710300156317,
            "mo_residual": 0.06989212918096037,
            "wall_time": 29.183796914
          },
          {
            "energy_terms": {
              "E_ee": 4.048829903319412,
              "E_eext": 0.0,
              "E_el": -7.4663514175705625,
              "E_en": -17.08949877826536,
              "E_kin": 7.381066252257522,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": 0.0,
              "E_tot": -7.4663514175705625,
              "E_x": -0.44311604114410696,
              "E_xc": -1.3636327537380306
            },
            "energy_total": -7.4663514175705625,
            "energy_update": 0.0008106592019663239,
            "mo_residual": 0.017564174517285668,
            "wall_time": 22.398756905
          }
        ],
        "wall_time": 51.583967276
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
