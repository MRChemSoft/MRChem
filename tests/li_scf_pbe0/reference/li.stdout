                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                                                                ***
***          __  __ ____   ____ _                                  ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \               ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |              ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|              ***
***                                                                ***
***         VERSION    0.0.1        (rev.   68df7e4c)              ***
***                                                                ***
***         Stig Rune Jensen   <stig.r.jensen@uit.no>              ***
***         Luca Frediani      <luca.frediani@uit.no>              ***
***         Peter Wind         <peter.wind@uit.no>                 ***
***                                                                ***
**********************************************************************

----------------------------------------------------------------------

 MPI processes         :                                            1
 OpenMP threads        :                                            1
 Total cores           :                                            1
                                                                      
----------------------------------------------------------------------

XCFun DFT library Copyright 2009-2011 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

----------------------------------------------------------------------

 MRCPP version         : X.Y.Z
 Git revision          : 316f21ec

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP

----------------------------------------------------------------------


                                                                      
======================================================================
                       MultiResolution Analysis
----------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
----------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                     Initializing Molecule                      ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            2
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0      Li          :       0.000000       0.000000       0.000000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000       0.000000
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Initial Guess Wavefunction              ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Reading Gaussian-type MOs
 Precision             : 1.00000e-03
 Restricted            : False
 Basis file            : initial_guess/mrchem.bas
 MO alpha file         : initial_guess/mrchem.moa
 MO beta file          : initial_guess/mrchem.mob
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                          Molecular Orbitals
----------------------------------------------------------------------
 Alpha electrons       :                                            2
 Beta electrons        :                                            1
 Total electrons       :                                            3
----------------------------------------------------------------------
    n  Occ Spin        :                                         Norm
----------------------------------------------------------------------
    0    1    a        :                           9.999999988178e-01
    1    1    a        :                           1.000000048122e+00
    2    1    b        :                           9.999999988146e-01
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***              Computing Ground State Wavefunction               ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Compute initial energy
 Method                : DFT
 Precision             : 1.00000e-03
 Localization          : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                7.370996808591
 E-N energy            :           (au)              -17.035030429630
 Coulomb energy        :           (au)                4.061917141109
 Exchange energy       :           (au)               -0.444799410662
 X-C energy            :           (au)               -1.366884780623
 Ext. field (el)       :           (au)                0.000000000000
 Ext. field (nuc)      :           (au)                0.000000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -7.413800671215
 Nuclear energy        :           (au)                0.000000000000
----------------------------------------------------------------------
 Total energy          :           (au)           -7.413800671215e+00
                       :     (kcal/mol)           -4.652230122466e+03
                       :       (kJ/mol)           -1.946493084503e+04
                       :           (eV)           -2.017397847484e+02
======================================================================


======================================================================
                           Orbital Energies
----------------------------------------------------------------------           
    n  Occ Spin        :                                      Epsilon            
----------------------------------------------------------------------           
    0    1    a        :           (au)               -2.026248624388            
    1    1    a        :           (au)               -0.139732410810            
    2    1    b        :           (au)               -2.015998440289            
----------------------------------------------------------------------           
 Sum occupied          :           (au)               -4.181979475487            
======================================================================



~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation           : Optimize molecular orbitals
 Method                : DFT
 Max iterations        : 5
 KAIN solver           : 3
 Localization          : Off
 Diagonalization       : First two iterations
 Start precision       : 1.00000e-03
 Final precision       : 1.00000e-03
 Energy threshold      : Off
 Orbital threshold     : 2.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


======================================================================
 Iter         MO residual                Property              Update
----------------------------------------------------------------------
    0        1.732051e+00         -7.413800671215       -7.413801e+00
    1        6.985284e-02         -7.465180151016       -5.137948e-02
    2        1.753844e-02         -7.466017735074       -8.375841e-04
----------------------------------------------------------------------
                   SCF converged in 2 iterations!
======================================================================


                                                                      
**********************************************************************
***                                                                ***
***                 Printing Molecular Properties                  ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
======================================================================
                               Molecule
----------------------------------------------------------------------
 Charge                :                                            0
 Multiplicity          :                                            2
----------------------------------------------------------------------
    N    Atom          :              x              y              z
----------------------------------------------------------------------
    0      Li          :       0.000000       0.000000       0.000000
----------------------------------------------------------------------
 Center of mass        :       0.000000       0.000000       0.000000
======================================================================


======================================================================
                           Orbital Energies
----------------------------------------------------------------------
    n  Occ Spin        :                                      Epsilon
----------------------------------------------------------------------
    0    1    a        :           (au)               -2.055885002544
    1    1    a        :           (au)               -0.141007077047
    2    1    b        :           (au)               -2.045656626704
----------------------------------------------------------------------
 Sum occupied          :           (au)               -4.242548706295
======================================================================


======================================================================
                           Molecular Energy
----------------------------------------------------------------------
 Kinetic energy        :           (au)                7.380314535068
 E-N energy            :           (au)              -17.088583572601
 Coulomb energy        :           (au)                4.048566358852
 Exchange energy       :           (au)               -0.442758246069
 X-C energy            :           (au)               -1.363556810324
 Ext. field (el)       :           (au)                0.000000000000
 Ext. field (nuc)      :           (au)                0.000000000000
----------------------------------------------------------------------
 Electronic energy     :           (au)               -7.466017735074
 Nuclear energy        :           (au)                0.000000000000
----------------------------------------------------------------------
 Total energy          :           (au)           -7.466017735074e+00
                       :     (kcal/mol)           -4.684996824481e+03
                       :       (kJ/mol)           -1.960202672635e+04
                       :           (eV)           -2.031606833792e+02
======================================================================


                                                                      
                                                                      
**********************************************************************
***                                                                ***
***                         Exiting MRChem                         ***
***                                                                ***
***                    Wall time :  0h  1m 11s                     ***
***                                                                ***
**********************************************************************
                                                                      
                                                                      
